docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+# 10101 Architecture
+
+This is a working document to design the architecture drafted in the below diagram. It will not remain in that structure but rather will be eventually documented in ADRs.
+
+![Assumed component diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/get10101/10101/d7619613324df3ce5c0f0a35d97bf24323f9515b/docs/diagrams/testing-component-diagram.puml)
+
+The following represents a non exhaustive list of topics relevant to be discussed when designing the 10101 architecture and is primarily based on the assumed component diagram introduced in https://github.com/get10101/10101/pull/30. Please also note that the order does not represent a priority of topics.
+
+## Database
+
+Both app and coordinator need to persist data. This section addresses the corresponding database decisions.
+
+### What data needs to be stored?
+
+ - Trading relevant data, like Offers, Orders, etc.
```

Comment:
I believe the term 'offer' will be phased out in favor of 'limited order' and 'market order'.

Reply:
In that case is the term "offer" a generalisation of "limited order" and "market order"?

Reply:
Mhmm... I would not make this generalisation. Offer is more a limited order from the point of view of the person who created the limited order but reserves the right to cancel it anytime.